attentive-android-sdk/src/androidTest/java/com/attentive/androidsdk/AttentiveApiTestIT.java

```diff
@@ -0,0 +1,268 @@
+package com.attentive.androidsdk;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.attentive.androidsdk.events.AddToCartEvent;
+import com.attentive.androidsdk.events.Cart;
+import com.attentive.androidsdk.events.Item;
+import com.attentive.androidsdk.events.Order;
+import com.attentive.androidsdk.events.Price;
+import com.attentive.androidsdk.events.ProductViewEvent;
+import com.attentive.androidsdk.events.PurchaseEvent;
+import com.attentive.androidsdk.internal.network.AddToCartMetadataDto;
+import com.attentive.androidsdk.internal.network.Metadata;
+import com.attentive.androidsdk.internal.network.ProductDto;
+import com.attentive.androidsdk.internal.network.ProductViewMetadataDto;
+import com.attentive.androidsdk.internal.network.PurchaseMetadataDto;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Currency;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+
+
+public class AttentiveApiTestIT {
+    private static final String DOMAIN = "mobileapps";
+    // Update this accordingly when running on VPN
+    private static final String GEO_ADJUSTED_DOMAIN = "mobileapps";
```

Comment:
we might want to change `GEO_ADJUSTED_DOMAIN` to have a different value than `DOMAIN` to ensure there's no false-positive equality checks

Reply:
chatted offline - testing from different locations is on our list of follow up items